src/binder/bind_node_visitor.cpp

```diff
@@ -495,6 +493,12 @@ void BindNodeVisitor::Visit(common::ManagedPointer<parser::SelectStatement> node
   node->SetSelectColumns(new_select_list);
   node->SetDepth(context_->GetDepth());
 
+  if (node->GetSelectOrderBy() != nullptr) {
+    auto order_by_expressions = UnifyOrderByExpression(node->GetSelectOrderBy(), node->GetSelectColumns());
+    node->GetSelectOrderBy()->GetOrderByExpressions().swap(order_by_expressions);
```

Comment:
I'm a little suspicious of this swap.

If the swap is necessary, then there should be new expressions being created, but I don't see them being added to the parse result.

If the swap is not necessary, because you took a reference to the original order by expressions and directly modified that, then the swap should probably be removed.

Reply:
@lmwnshn thank you for your suggestion. And I have removed the swap and modified the UnifyOrderByExpression function.